src/python/integration/pip_test.go

```diff
@@ -0,0 +1,226 @@
+
+						Eventually(deployment).Should(Serve("Hello, World!"))
+
+						Expect(logs.String()).To(SatisfyAll(
+							ContainSubstring("Using python's pip module"),
+						))
+					})
+				})
+
+				context("when using latest version", func() {
+					it.Before(func() {
+						var err error
+						source, err = switchblade.Source(filepath.Join(fixtures, "simple"))
+						Expect(err).NotTo(HaveOccurred())
+					})
+
+					it.After(func() {
+						Expect(os.RemoveAll(source)).To(Succeed())
+					})
+
+					it("uses latest from manifest", func() {
+						deployment, logs, err := platform.Deploy.
+							WithEnv(map[string]string{"BP_PIP_VERSION": "latest"}).
+							Execute(name, source)
+						Expect(err).NotTo(HaveOccurred())
+
+						Eventually(deployment).Should(Serve("Hello, World!"))
+
+						Expect(logs.String()).To(SatisfyAll(
+							ContainSubstring("Installing pip"),
+							MatchRegexp(`Successfully installed pip-\d+.\d+(.\d+)?`),
+						))
+					})
+				})
+			})
+		})
+
+		context("when there is no requirements.txt file", func() {
+			context("when there is a setup.py file", func() {
+				it.Before(func() {
+					var err error
+					source, err = switchblade.Source(filepath.Join(fixtures, "setup_py"))
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				it.After(func() {
+					Expect(os.RemoveAll(source)).To(Succeed())
+				})
+
+				it("successfully deploys", func() {
+					deployment, logs, err := platform.Deploy.
+						WithBuildpacks("python_buildpack").
+						Execute(name, source)
+					Expect(err).NotTo(HaveOccurred())
+
+					Eventually(deployment).Should(Serve("Knock Knock. Who is there?"))
+
+					Expect(logs.String()).To(SatisfyAll(
+						Not(ContainSubstring("Skipping 'pip install' since requirements.txt does not exist")),
+					))
+				})
+			})
+
+			context("when there is no setup.py file", func() {
+				it.Before(func() {
+					var err error
+					source, err = switchblade.Source(filepath.Join(fixtures, "no_deps"))
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				it.After(func() {
+					Expect(os.RemoveAll(source)).To(Succeed())
+				})
+
+				it("successfully deploys", func() {
+					deployment, logs, err := platform.Deploy.
+						WithBuildpacks("python_buildpack").
+						Execute(name, source)
+					Expect(err).NotTo(HaveOccurred())
+
+					Eventually(deployment).Should(Serve("Here is your output for /"))
+
+					Expect(logs.String()).To(SatisfyAll(
+						Not(ContainSubstring("Skipping 'pip install' since requirements.txt does not exist")),
+					))
+				})
+			})
+		})
+	}
+}
+
+func CreateRequirementsTxtFile(Expect func(actual interface{}, extra ...interface{}) types.Assertion, path string, fileName string, modules ...string) {
+	file, err := os.Create(filepath.Join(path, fileName))
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, module := range modules {
+		_, err = file.WriteString(module + "\n")
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	err = file.Close()
+
+	Expect(err).NotTo(HaveOccurred())
+}
```

Comment:
I might suggest moving this to `init_test.go` since it is used across more than just this test file.

Reply:
Moved, thanks!